aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/HttpChecksumTest.kt

```diff
@@ -427,6 +460,128 @@ internal class HttpChecksumTest {
             )
         }
     }
+
+    /**
+     * Generate miscellaneous tests
+     */
+    private fun createMiscellaneousTests(context: ClientCodegenContext): Writable {
```

Comment:
Great addition! Can this function somehow use `test-util` only [assert_ua_contains_metric_values](https://github.com/smithy-lang/smithy-rs/blob/abd28bc71ee91a3e59b8afc801364b909436bb94/aws/rust-runtime/aws-runtime/src/user_agent/test_util.rs#L19) (or improve it if necessary)?

Reply:
Good call, missed those test-utils earlier. Updated to use them in https://github.com/smithy-lang/smithy-rs/pull/3851/commits/d714eaa67f02859192622eefa309e5facfca81b0.

Refactored the test-util slightly and split out the extraction from the assertion since I wanted access to the actual underlying items. This is so that I could assert on the absence of some UA headers (the headers for the various checksum configs should be mutually exclusive), but that use case seems uncommon enough that it probably doesn't warrant its own util.

Reply:
Cool refactoring!